plugins/feeds/public/firehol_github.py

```diff
@@ -0,0 +1,75 @@
+import logging
+from dateutil import parser
+from datetime import datetime, timedelta
+from core.feed import Feed
+from core.errors import GenericYetiError
+from core.observables import Observable
+from core.observables import Hash, Url, Hostname, Ip, MacAddress, Email, Certificate
+from core.errors import ObservableValidationError
+from core.config.config import yeti_config
+
+
+class FireHolGitHub(Feed):
+
+    default_values = {
+        'frequency': timedelta(hours=24),
+        'name': 'FireHolGitHub',
+        'source': 'https://api.github.com/repos/firehol/blocklist-ipsets/commits',
+        'description': 'Get Iocs from FireHol repo',
+    }
+    refs = {
+        'MacAddress': MacAddress,
+        'Hash': Hash,
+        'Url': Url,
+        'Ip': Ip,
+        'FileHash-SHA1': Hash,
+        'Hostname': Hostname,
+        'Email': Email,
+    }
+
+    blacklist = ('.gitignore', 'LICENSE', 'README.adoc', 'README-EDIT.md', 'README.md')
+    blacklist_domains = (
+        'technet.microsoft.com', 'cloudblogs.microsoft.com', 'capec.mitre.org',
+        'attack.mitre.org', 'securelist.com', 'blog.avast.com', 'firehol.org',
+        'gist.githubusercontent.com', 'www.binarydefense.com', 'www.badips.com')
+
+    def process_content(self, content, filename):
+        context = dict(source=self.name)
+        context['description'] = 'File: {}'.format(filename)
+
+        if content.startswith('Certificate:') and content.endswith(
+                '-----END CERTIFICATE-----\n'):
+
+            try:
+                cert_data = Certificate.from_data(content)
```

Comment:
Please refactor these blocks into methods (one for certificates, the other for observables)